src/test/java/edu/harvard/iq/dataverse/api/DatasetsIT.java

```diff
@@ -2583,5 +2583,48 @@ public void testFilesUnchangedAfterDatasetMetadataUpdate() throws IOException {
                 .body("data.latestVersion.files[0].directoryLabel", equalTo("code"));
         
     }
+
+    @Test
+    public void testAddFileToDatasetTabIngest() throws IOException, InterruptedException {
+
+        Response createUser = UtilIT.createRandomUser();
+        assertEquals(200, createUser.getStatusCode());
+        String username = UtilIT.getUsernameFromResponse(createUser);
+        String apiToken = UtilIT.getApiTokenFromResponse(createUser);
+
+        Response createDataverseResponse = UtilIT.createRandomDataverse(apiToken);
+        assertEquals(201, createDataverseResponse.getStatusCode());
+        String dataverseAlias = UtilIT.getAliasFromResponse(createDataverseResponse);
+
+        Response createDatasetResponse = UtilIT.createRandomDatasetViaNativeApi(dataverseAlias, apiToken);
+        assertEquals(201, createDatasetResponse.getStatusCode());
+        Integer datasetIdInt = JsonPath.from(createDatasetResponse.body().asString()).getInt("data.id");
+
+        String pathToFile = "src/test/resources/sav/dct.sav";
+        String jsonAsString = "{\"description\":\"My description.\",\"directoryLabel\":\"data/subdir1\",\"categories\":[\"Data\"], \"restrict\":\"false\", \"tabIngest\":\"false\"}";
+        Response r = UtilIT.uploadFileViaNative(datasetIdInt.toString(), pathToFile, jsonAsString, apiToken);
+        logger.info(r.prettyPrint());
+        assertEquals(200, r.getStatusCode());
+
+        pathToFile = "src/test/resources/sav/frequency-test.sav";
+        jsonAsString = "{\"description\":\"My description.\",\"directoryLabel\":\"data/subdir1\",\"categories\":[\"Data\"], \"restrict\":\"false\"  }";
+        Response rTabIngest = UtilIT.uploadFileViaNative(datasetIdInt.toString(), pathToFile, jsonAsString, apiToken);
+        logger.info(rTabIngest.prettyPrint());
+        assertEquals(200, rTabIngest.getStatusCode());
```

Comment:
Checking the file name may be enough (making sure it's still ".sav", and not ".tab").

Reply:
But yes, I agree, we do need to confirm that the ingest was indeed skipped.
Also, the 2nd file upload in this method - where ingest is NOT being skipped - it may not be necessary, since we are already testing actual ingest elsewhere (in FilesIT.java?).

Reply:
Ok, I renamed the test and moved it to FilesIT. I also added the ingest check by checking label of file metadata.